chia/wallet/nft_wallet/metadata_outer_puzzle.py

```diff
@@ -45,8 +43,8 @@ def match(self, puzzle: UncurriedPuzzle) -> Optional[PuzzleInfo]:
             _, metadata, updater_hash, inner_puzzle = curried_args
             constructor_dict = {
                 "type": "metadata",
-                "metadata": disassemble(metadata),
-                "updater_hash": "0x" + updater_hash.as_python().hex(),
+                "metadata": metadata,
```

Comment:
It's great to not have to disassemble. But I was under the impression this field *had* to be disassembled.
tracing this, it looks like we end up passing this metadata into `puzzle_for_metadata_layer()` which has taken a `Program` all along. I don't understand why it was correct to `disassemble()` this to begin with. I would expect there to be a corresponding `assemble()` call removed somewhere.

Reply:
I agree that this _should_ be disassembled because other calls generally to this API will not get the right output.  PuzzleInfo is just bad at type checking itself and if it was better, it would likely throw an error due to this change.  However, in context, I would sort of like to deprecate this code (or make `PuzzleInfo` better so that this would actually not be a problem) so if all of the calls to this method still work, I'm actually fine with changing this.

Reply:
I see you changed `decode_info_value` to be able to handle this actually so that's perfect :+1:

Reply:
We need to move away from using `disassemble()` and `assemble()` in general

Reply:
I agree in the context of such a haphazard manner, but some RPCs need to accept/return CLVM them and there's really no way around those methods in that case.